From 1c89d567b594af44eb18b28ebe1df5ae18f8b032 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Fri, 18 Jul 2014 02:23:10 +1000
Subject: [PATCH 1/3] metadata: fix wrong test leading to incorrect hash
 calculation

This would have almost always returned true, meaning it would attempt to calculate the hash of a file unlikely to exist locally.
Simplify code + style
---
 mythtv/libs/libmythmetadata/videometadata.cpp |   17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/mythtv/libs/libmythmetadata/videometadata.cpp b/mythtv/libs/libmythmetadata/videometadata.cpp
index 4dabcd3..309e271 100644
--- a/mythtv/libs/libmythmetadata/videometadata.cpp
+++ b/mythtv/libs/libmythmetadata/videometadata.cpp
@@ -1066,19 +1066,20 @@ int VideoMetadata::UpdateHashedDBRecord(const QString &hash,
 QString VideoMetadata::VideoFileHash(const QString &file_name,
                            const QString &host)
 {
-    if (!host.isEmpty() && !gCoreContext->IsMasterHost(host))
-    {
-        QString url = generate_file_url("Videos", host, file_name);
-        return RemoteFile::GetFileHash(url);
-    }
-    else if (!host.isEmpty())
+    if (host.isEmpty())
+        return FileHash(file_name);
+
+    if (gCoreContext->IsMasterBackend() && gCoreContext->IsThisHost(host))
     {
         StorageGroup sgroup("Videos", host);
         QString fullname = sgroup.FindFile(file_name);
+
         return FileHash(fullname);
     }
-    else
-        return FileHash(file_name);
+
+    QString url = generate_file_url("Videos", host, file_name);
+
+    return RemoteFile::GetFileHash(url);
 }
 
 QString VideoMetadata::FilenameToMeta(const QString &file_name, int position)
-- 
1.7.10.2

